Failing Nx affected:lint does not cause GitLab CI step to fail

Hi,

we are currently experiencing some difficulties with the integration of Nx lint into our existing GitLab CI setup. The problem we are having, is that our CI step does not fail, even though there are clearly linting issues and by all means it should fail.

Our CI currently runs this command:
`./node_modules/.bin/nx affected:lint --base=master --parallel`

Below is an excerpt of the output I see from our CI runner (there are many more errors available, from all different libraries in the repository). Of especial importance is the final `Job succeeded`, as this indicates the CI step was completed correctly.

```
ERROR: /builds/libs/ui-utils/src/directives/sort/sort.directive.spec.ts[359, 36]: Type declaration of 'any' loses type-safety. Consider replacing it with a more precise type.
ERROR: /builds/libs/ui-utils/src/directives/sort/sort.directive.spec.ts[403, 36]: Type declaration of 'any' loses type-safety. Consider replacing it with a more precise type.
ERROR: /builds/libs/ui-utils/src/directives/sort/sort.directive.spec.ts[465, 36]: Type declaration of 'any' loses type-safety. Consider replacing it with a more precise type.
ERROR: /builds/libs/ui-utils/src/directives/sort/sort.directive.spec.ts[490, 36]: Type declaration of 'any' loses type-safety. Consider replacing it with a more precise type.
ERROR: /builds/libs/ui-utils/src/directives/sort/sort.directive.spec.ts[540, 36]: Type declaration of 'any' loses type-safety. Consider replacing it with a more precise type.
ERROR: /builds/libs/ui-utils/src/directives/sort/sort.directive.ts[24, 26]: Type declaration of 'any' loses type-safety. Consider replacing it with a more precise type.
ERROR: /builds/libs/ui-utils/src/directives/sort/sort.directive.ts[81, 13]: Type declaration of 'any' loses type-safety. Consider replacing it with a more precise type.
ERROR: /builds/libs/ui-utils/src/directives/sort/sort.directive.ts[86, 11]: Type declaration of 'any' loses type-safety. Consider replacing it with a more precise type.
ERROR: /builds/libs/ui-utils/src/directives/sort/sort.directive.ts[163, 38]: Type declaration of 'any' loses type-safety. Consider replacing it with a more precise type.
ERROR: /builds/libs/ui-utils/src/directives/sort/sort.directive.ts[207, 31]: Type declaration of 'any' loses type-safety. Consider replacing it with a more precise type.
ERROR: /builds/libs/ui-utils/src/utils/compare/compare.util.ts[14, 40]: Type declaration of 'any' loses type-safety. Consider replacing it with a more precise type.
ERROR: /builds/libs/ui-utils/src/utils/compare/compare.util.ts[14, 49]: Type declaration of 'any' loses type-safety. Consider replacing it with a more precise type.
ERROR: /builds/libs/users/src/+state/user-groups/user-groups.actions.ts[18, 42]: Type declaration of 'any' loses type-safety. Consider replacing it with a more precise type.
ERROR: /builds/libs/users/src/+state/users/users.actions.ts[18, 42]: Type declaration of 'any' loses type-safety. Consider replacing it with a more precise type.
ERROR: /builds/libs/users/src/components/groups-table/groups-table.component.ts[48, 19]: Type declaration of 'any' loses type-safety. Consider replacing it with a more precise type.
ERROR: /builds/libs/users/src/components/users-table/users-table.component.ts[53, 18]: Type declaration of 'any' loses type-safety. Consider replacing it with a more precise type.


Lint errors found in the listed files.
npm ERR! code ELIFECYCLE
npm ERR! errno 1
npm ERR! op-frontend@0.0.0 ng: `ng "lint" "--project=op-frontend"`
npm ERR! Exit status 1
npm ERR! 
npm ERR! Failed at the op-frontend@0.0.0 ng script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.

npm ERR! A complete log of this run can be found in:
npm ERR!     /root/.npm/_logs/2018-08-22T14_34_28_384Z-debug.log
(node:30) UnhandledPromiseRejectionWarning: Error: ENOENT: no such file or directory, open 'dist/.nx-results'
    at Object.openSync (fs.js:434:3)
    at Object.writeFileSync (fs.js:1154:35)
    at writeToFile (/usr/src/app/node_modules/@nrwl/schematics/src/utils/fileutils.js:7:8)
    at Object.writeJsonFile (/usr/src/app/node_modules/@nrwl/schematics/src/utils/fileutils.js:61:5)
    at WorkspaceResults.saveResults (/usr/src/app/node_modules/@nrwl/schematics/src/command-line/workspace-results.js:64:25)
    at /usr/src/app/node_modules/@nrwl/schematics/src/command-line/affected.js:247:38
    at step (/usr/src/app/node_modules/@nrwl/schematics/src/command-line/affected.js:40:23)
    at Object.throw (/usr/src/app/node_modules/@nrwl/schematics/src/command-line/affected.js:21:53)
    at rejected (/usr/src/app/node_modules/@nrwl/schematics/src/command-line/affected.js:13:65)
    at process._tickCallback (internal/process/next_tick.js:68:7)
(node:30) UnhandledPromiseRejectionWarning: Unhandled promise rejection. This error originated either by throwing inside of an async function without a catch block, or by rejecting a promise which was not handled with .catch(). (rejection id: 1)
(node:30) [DEP0018] DeprecationWarning: Unhandled promise rejections are deprecated. In the future, promise rejections that are not handled will terminate the Node.js process with a non-zero exit code.
**Job succeeded**
```

Is this known and/or desired behaviour?

This is currently limiting our adoption of Nx, as we don't have a linting enforced in our CI setup right now. When running `ng lint` we get the correct behaviour, but for obvious performance reasons this is not how we want to configure our monorepo.

Do I have to add a certain flag to the command maybe, to get the `affected:lint` to correctly fail? If so, why would it not fail immediately?

Thanks for the help!
Nils

--- a/packages/schematics/src/utils/fileutils.ts
+++ b/packages/schematics/src/utils/fileutils.ts
@@ -1,9 +1,11 @@
 import { statSync } from 'fs';
 import * as fs from 'fs';
 import * as path from 'path';
+import { ensureDirSync } from 'fs-extra';
 
-export function writeToFile(path: string, str: string) {
-  fs.writeFileSync(path, str);
+export function writeToFile(filePath: string, str: string) {
+  ensureDirSync(path.dirname(filePath));
+  fs.writeFileSync(filePath, str);
 }
 
 /**

